requests-for-comments/2022-12-01-workflow-new-feature.md

```diff
@@ -0,0 +1,57 @@
+# The first "Workflow": MVP for "Creating a new feature"
+
+Owner: @EDsCODE (team experimentation)
+Decision to be made: Should we build this in Q1 as part of team experimentation?
+Decision date: 7th December 2022
+
+## Context
+
+This RFC proposes that team experimentation builds the MVP of a workflow for "Creating a new feature" as part of their Q1 OKRs.
+
+It should sit with team experimentation as it would be a key driver of feature flag usage.
+
+This could have multiple benefits across the product:
+
+- Making PostHog the first place that product engineers go when building a new feature
+- Increasing cross-product usage (feature flags, product analytics, session recordings)
+- Encoding the best practices for PostHog and external engineers to use to build check for feature success without using PM time
+
+*This is context for the engineers leading it and should be iterated*
+
+## What's a workflow?
+
+So far we have given product engineers the individual tools to create trends, create funnels, watch recordings, create feature flags etc.
+
+We have the opportunity to connect these flexible tools with opinionated best practices to focus on specific jobs-to-be-dones in PostHog. These could be called "Workflows".
+
+Two examples of potential workflows:
+
+1. Building a new feature - assisting with rollout and measuring success *More details below*
+2. Workbooks - making it easy to come up with questions, save related insights, clip particular bits of recordings and jot down thoughts
+
+## Vision for "Creating a new feature" workflow
+
+Here's the vision of what this could be in 3 - 6 months:
+
+Sarah is a product engineer at a high-growth startup building a B2C website to help companies offset their carbon - called Carbon.io. She's working on a new feature for users to watch videos of their carbon-offsetting projects. She hypothesizes that users who watch these videos will feel more connected to the projects and retain at a higher rate.
+
+The first thing Sarah does is open up PostHog and start the workflow for "Create a new feature". It opens up a page to create the new feature where she names it "Videos of offsetting projects". Optionally, she can enter some of the background information on this feature including the hypothesis. She selects "auto-rollout" rather than "A/B test" - meaning once the feature is made it will be used internally and after manual approval will be rolled out to beta users then low-value users then high-value users. She selects "user retention" as the key metric.
```

Comment:
Do we think this feature will be applicable for existing users or are we hoping to attract a new set of users? If existing, is there evidence that current users would be interested in working this way. For example, are event taxonomies and feature flags decided upon pre feature implementation or post?

If not existing, is there some research we can do to get information on how this plays against the existing popular workflows among product engineers? I would assume there's some level of tooling or assortment of tools that are used to achieve this product planning -> implementation